src/lib.rs

```diff
@@ -52,6 +52,25 @@ pub struct DateTime {
     pub time: Time,
 }
 
+/// A duration, can hold three different formats.
+#[derive(Eq, PartialEq, Debug, Copy, Clone)]
+pub enum Duration {
+    /// consists of year, month, day, hour, minute and second units
+    YMDHMS {
+        year: u32,
+        month: u32,
+        day: u32,
+        hour: u32,
+        minute: u32,
+        second: u32,
+        millisecond: u32,
+    },
+    /// consists of week units
+    Weeks(u32),
+    /// consists of an ISO 8601 format [DateTime](struct.DateTime.html)
+    DateTime(DateTime),
```

Comment:
What even is a duration in a datetime format? o_O

Reply:
Ooooh, rereading the wikipedia article now this is to be interpreted similar to the `YMDHMS` format (at least if communicating parties agree).
Can't we therefore just convert it to the above to remove ambiguity?

Reply:
(and document what and why we did it)

Reply:
Done and added some documentation, let me know if it's sufficient! 😄